### ISSUE TITLE
Observer with wildcards fire unexpectedly when updating arrays at indexes > 9

### ISSUE THREAD

**Issue by giovannipiller** [ractivejs/ractive#2805]
## Description:
Found another bug on observers with wildcards.
Setting values of elements at indexes bigger than 9 will result in the observer firing twice.

Example:
setting `container.0.list.11`: will fire a change for `container.0.list.1` then another for `container.0.list.11`.

Similarly, setting `container.0.list.22`: will fire a change for `container.0.list.2` then another for `container.0.list.22`.

## Versions affected:
0.8.x, Edge

## Reproduction:
https://jsfiddle.net/gigasoft/x3cj9u9x/1/

## Code:
```
const r = window.r = new Ractive({
  el: '#main',
  template: '#template',
  data: {

    container: [{
      list: [
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
        {test: {within: 1}},
      ]
    }]
  },
  oninit() {
    this.observe('container.*.list.*', (newValue, oldValue, keypath, container, list) => {
      // will fire twice, the first time pointing at container.0.list.1
      console.info(keypath, container, list)
    }, {init: false})
  },
  onrender() {
    this.set('container.0.list.10.test.within', (new Date()).getTime());
  }
});
```

**Comment by ghost** [comment]
Verified that it works with 0.7.3 https://jsfiddle.net/35tnjv4n/.

**Comment by evs-chris** [comment]
The issue is that pattern observers are checking keypaths for overlap but not _exact_ overlap, so `foo.ba` overlaps with `foo.bar.baz`, but not exactly. The fix is pretty simple, and I have a commit ready to push that fixes it, but it looks like one of our build deps is broken for the 0.8 branch.

**Comment by evs-chris** [comment]
This is now fixed in 0.8-dev and edge, and will be published when 0.8.8 is released. Thanks for the report!
